src/ShopBundle/objects/db/TShopCategory.class.php

```diff
@@ -648,8 +648,17 @@ public function getTargetPage()
             if (null === $activePortal) {
                 return null;
             }
-            $targetPageId = $activePortal->GetSystemPageId('products');
-            $defaultPage = TdbCmsTplPage::GetNewInstance($targetPageId);
+
+            $systemPageService = $this->getSystemPageService();
+            $defaultSystemPage = $systemPageService->getSystemPage('products', $activePortal);
+
+            if (null === $defaultSystemPage) {
+                return null;
+            }
+
+            $pageService = self::getPageService();
+            $defaultPage = $pageService->getById($defaultSystemPage->id);
```

Comment:
Could directly set $targetpage.

Reply:
No, because $defaultPage is a static variable in this method. By not changing it as we did before we change the behavior of this method on consecutive calls.